sdk/communication/Azure.Communication.Rooms/src/RoomsClient.cs

```diff
@@ -0,0 +1,511 @@
+                new RoomsClientOptions())
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
+        /// </summary>
+        /// <param name="connectionString"></param>
+        /// <param name="options"></param>
+        public RoomsClient(string connectionString, RoomsClientOptions options)
+            : this(
+                ConnectionString.Parse(Argument.CheckNotNullOrEmpty(connectionString, nameof(connectionString))),
+                options ?? new RoomsClientOptions())
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="keyCredential"></param>
+        /// <param name="options"></param>
+        public RoomsClient(Uri endpoint, AzureKeyCredential keyCredential, RoomsClientOptions options = default)
+            : this(
+                Argument.CheckNotNull(endpoint, nameof(endpoint)).AbsoluteUri,
+                Argument.CheckNotNull(keyCredential, nameof(keyCredential)),
+                options ?? new RoomsClientOptions())
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="tokenCredential"></param>
+        /// <param name="options"></param>
+        public RoomsClient(Uri endpoint, TokenCredential tokenCredential, RoomsClientOptions options = default)
+            : this(
+                Argument.CheckNotNull(endpoint, nameof(endpoint)).AbsoluteUri,
+                Argument.CheckNotNull(tokenCredential, nameof(tokenCredential)),
+                options ?? new RoomsClientOptions())
+        { }
+
+        #endregion
+
+        #region private constructors
+
+        private RoomsClient(ConnectionString connectionString, RoomsClientOptions options)
+            : this(connectionString.GetRequired("endpoint"), options.BuildHttpPipeline(connectionString), options)
+        { }
+
+        private RoomsClient(string endpoint, TokenCredential tokenCredential, RoomsClientOptions options)
+            : this(endpoint, options.BuildHttpPipeline(tokenCredential), options)
+        { }
+
+        private RoomsClient(string endpoint, AzureKeyCredential keyCredential, RoomsClientOptions options)
+            : this(endpoint, options.BuildHttpPipeline(keyCredential), options)
+        { }
+
+        private RoomsClient(string endpoint, HttpPipeline httpPipeline, RoomsClientOptions options)
+        {
+            _clientDiagnostics = new ClientDiagnostics(options);
+            RoomsServiceClient = new RoomsRestClient(_clientDiagnostics, httpPipeline, endpoint, options.ApiVersion);
+        }
+        #endregion
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
```

Comment:
Nit - add `for mocking.` on the end of the comment to help customers who see this in our ref docs.

Reply:
Addressed.